minikube addons configure on a nonexistent cluster: panic: runtime error

To reproduce:

2. `minikube -p xyz addons configure metallb`

Example output:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x288 pc=0x224dd7a]

goroutine 1 [running]:
k8s.io/minikube/cmd/minikube/cmd/config.glob..func8(0x37153a0, 0xc000275350, 0x1, 0x1)
	/app/cmd/minikube/cmd/config/configure.go:200 +0x1d2a
github.com/spf13/cobra.(*Command).execute(0x37153a0, 0xc000275320, 0x1, 0x1, 0x37153a0, 0xc000275320)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0x3712ee0, 0x0, 0x1, 0xc0004156c0)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
k8s.io/minikube/cmd/minikube/cmd.Execute()
	/app/cmd/minikube/cmd/root.go:112 +0x747
main.main()
	/app/cmd/minikube/main.go:66 +0xea
```

--- a/cmd/minikube/cmd/config/configure.go
+++ b/cmd/minikube/cmd/config/configure.go
@@ -23,6 +23,7 @@ import (
 	"github.com/spf13/cobra"
 	"k8s.io/minikube/pkg/minikube/config"
 	"k8s.io/minikube/pkg/minikube/exit"
+	"k8s.io/minikube/pkg/minikube/mustload"
 	"k8s.io/minikube/pkg/minikube/out"
 	"k8s.io/minikube/pkg/minikube/service"
 )
@@ -35,6 +36,8 @@ var addonsConfigureCmd = &cobra.Command{
 		if len(args) != 1 {
 			exit.UsageT("usage: minikube addons configure ADDON_NAME")
 		}
+
+		mustload.Partial(ClusterFlagValue())
 
 		addon := args[0]
 		// allows for additional prompting of information when enabling addons

--- a/cmd/minikube/cmd/config/configure.go
+++ b/cmd/minikube/cmd/config/configure.go
@@ -37,8 +37,6 @@ var addonsConfigureCmd = &cobra.Command{
 			exit.UsageT("usage: minikube addons configure ADDON_NAME")
 		}
 
-		mustload.Partial(ClusterFlagValue())
-
 		addon := args[0]
 		// allows for additional prompting of information when enabling addons
 		switch addon {
@@ -191,10 +189,7 @@ var addonsConfigureCmd = &cobra.Command{
 
 		case "metallb":
 			profile := ClusterFlagValue()
-			cfg, err := config.Load(profile)
-			if err != nil {
-				out.ErrT(out.FatalType, "Failed to load config {{.profile}}", out.V{"profile": profile})
-			}
+			_, cfg := mustload.Partial(profile)
 
 			validator := func(s string) bool {
 				return net.ParseIP(s) != nil
@@ -208,8 +203,7 @@ var addonsConfigureCmd = &cobra.Command{
 				cfg.KubernetesConfig.LoadBalancerEndIP = AskForStaticValidatedValue("-- Enter Load Balancer End IP: ", validator)
 			}
 
-			err = config.SaveProfile(profile, cfg)
-			if err != nil {
+			if err := config.SaveProfile(profile, cfg); err != nil {
 				out.ErrT(out.FatalType, "Failed to save config {{.profile}}", out.V{"profile": profile})
 			}
 

